larana/OpticalDetector/OpHitFinder/RiseTimeTools/RiseTimeGaussFit_tool.cc

```diff
@@ -0,0 +1,136 @@
+/**
+ * \file RiseTimeGaussFit_tool.cc
+ *
+ * \brief Gaussian fit method implemented to  compute Rise time as
+ * the value of the center of the first local maximum.
+ * Fixed min threshold required set in the fhicl file
+ *
+ * @author Rodrigo Alvarez, Sept 2022
+ */
+
+#include "art/Utilities/ToolConfigTable.h"
+#include "art/Utilities/ToolMacros.h"
+#include "fhiclcpp/types/Atom.h"
+
+#include "RiseTimeCalculatorBase.h"
+
+// ROOT includes
+#include "TF1.h"
+#include "TH1F.h"
+
+namespace pmtana {
+
+  class RiseTimeGaussFit : RiseTimeCalculatorBase {
+
+  public:
+    //Configuration parameters
+    struct Config {
+
+      fhicl::Atom<double> MinAmp{fhicl::Name("MinAmp")};
+      fhicl::Atom<double> InitSigma{fhicl::Name("InitSigma")};
+      fhicl::Atom<int> Nbins{fhicl::Name("Nbins")};
+      fhicl::Atom<double> Tolerance{fhicl::Name("Tolerance")};
+    };
+
+    // Default constructor
+    explicit RiseTimeGaussFit(art::ToolConfigTable<Config> const& config);
+
+    // Method to calculate the OpFlash t0
+    double RiseTime(const pmtana::Waveform_t& wf_pulse,
+                    const pmtana::PedestalMean_t& ped_pulse,
+                    bool _positive) const override;
+    // Method to fit the first local max of the wvf above fixed threshold
+    int findFirstMax(const std::vector<double>& arr, int low, int high, double threshold) const;
```

Comment:
As `findFirstMax` is used only to iterate over the entire array, I suggest removing the `low` and `high` arguments:

```C++
std::size_t findfirstMax(const std::vector<double>& arr, double threshold) const;
```